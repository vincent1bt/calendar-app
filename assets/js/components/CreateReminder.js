import React from 'react';
import PropTypes from 'prop-types';

const CreateReminder = ({ onReminderClick, eventId, monthId }) => {
  let input;
  return (
    <div className="reminder-input">
      <input ref={node => {
        input = node
      }} />
      <button onClick={() => {
        onReminderClick(input.value, eventId, monthId)
        input.value = ''
      }}>
      Add
      </button>
    </div>
  );
}

CreateReminder.propTypes = {
  onReminderClick: PropTypes.func.isRequired,
  eventId: PropTypes.number.isRequired,
  monthId: PropTypes.string.isRequired
}

export default CreateReminder;
