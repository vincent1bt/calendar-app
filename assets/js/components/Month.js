import React from 'react';
import PropTypes from 'prop-types';

const Month = ({ list, name, year }) => {
  if(list !== null) {
    return (
      <div className="month">
        <h2>{name} {year}</h2>
        <ul className="month-names">
          <li><p>Mon</p></li>
          <li><p>Tue</p></li>
          <li><p>Wed</p></li>
          <li><p>Thu</p></li>
          <li><p>Fri</p></li>
          <li><p>Sat</p></li>
          <li><p>Sun</p></li>
        </ul>
        <ul className="month-days">
          {list}
        </ul>
      </div>
    )
  } else {
    return (
      <div className="month">
        <div className="loader">
          <img src="/static/spin.svg" />
        </div>
      </div>
    )
  }
}

Month.propTypes = {
  list: PropTypes.array,
  name: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired
}

export default Month;
