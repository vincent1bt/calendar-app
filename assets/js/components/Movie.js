import React from 'react';

const Movie = ({ movie }) => {
  return (
    <div className="movie">
        <div className="title">
          <h3 className="title-p" >{movie.get("title")}</h3>
          <p className="title-over" >{movie.get("overview")}</p>
        </div>
        <div className="image">
          <img src={`http://image.tmdb.org/t/p/w500${movie.get("poster_path")}`} alt={movie.get("title")} />
        </div>
    </div>
  );
}

export default Movie;
