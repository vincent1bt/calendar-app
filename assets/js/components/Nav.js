import React from 'react';
import { NavLink } from 'react-router-dom';

const months = [
  "January", "February", "March",
  "April", "May", "June", "July",
  "August", "September", "Octuber",
  "November", "December"
];

const Nav = () => {
  return (
    <nav className="months-nav">
      <ul className="links">
        {
          months.map((month, index) => {
            return (
              <li key={index} >
                <NavLink to={`/month/${index + 1}`}
                      activeClassName="activeMonthLink">
                </NavLink>
                {month}
              </li>
            )
          })
        }
      </ul>
    </nav>
  )
};

export default Nav;
