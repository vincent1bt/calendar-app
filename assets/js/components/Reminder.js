import React from 'react';

const Reminder = ({ reminder }) => {
  return (
    <div className="reminder">
        <p>{reminder.get("text")}</p>
    </div>
  );
}

export default Reminder;
