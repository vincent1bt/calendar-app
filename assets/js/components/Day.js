import React, { PropTypes } from 'react';
import CreateReminder from 'components/CreateReminder';
import getEvents from 'helpers/getEvents';

const Day = ({ monthName, monthId, eventId, dayId, events, reminderClick }) => {
  const event = getEvents(events);
  return (
    <div>
      <section className="titles-day">
        <h2>Day {dayId}, {monthName}</h2>
      </section>
      <section className="movies">
        <h2 className="movies-title">Movies</h2>
        {event.movies}
      </section>
      <section className="reminders">
        <h2 className="reminders-title">Reminders</h2>
        {event.reminders}
        <CreateReminder onReminderClick={reminderClick} eventId={eventId} monthId={monthId} />
      </section>
    </div>
  )
}

Day.propTypes =  {
  monthName: PropTypes.string.isRequired,
  monthId: PropTypes.string.isRequired,
  eventId: PropTypes.number.isRequired,
  dayId: PropTypes.string.isRequired,
  events: PropTypes.object,
  reminderClick: PropTypes.func.isRequired
}
// TodoList.propTypes = {
//   todos: PropTypes.arrayOf(PropTypes.shape({
//     id: PropTypes.number.isRequired,
//     completed: PropTypes.bool.isRequired,
//     text: PropTypes.string.isRequired
//   }).isRequired).isRequired,
//   onTodoClick: PropTypes.func.isRequired
// };

export default Day;
