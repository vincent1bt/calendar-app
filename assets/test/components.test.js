import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';

import Month from 'components/Month';
import Day from 'components/Day';
import Movie from 'components/Movie';
import Reminder from 'components/Reminder';
import CreateReminder from 'components/CreateReminder';
import Nav from 'components/Nav';


describe('components', () => {
  describe('<Month/>', () => {
    it('renders', () => {
      const tree = renderer.create(<Month name="enero" year="2017" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("<Day/>", () => {
    it('renders', () => {
      const tree = renderer.create(
          <Day
            monthName="enero"
            monthId="1"
            eventId={345}
            dayId="12"
            reminderClick={ () => null }
          />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("<Movie/>", () => {
    it('renders', () => {
      const movie = {
        id: 339967,
        overview: "A woman discovers that severe catastrophic events are somehow connected to the mental breakdown from which she's suffering.",
        poster_path: "/4VOyofBd1pexblxtDZYtYIk7NI4.jpg",
        release_day: "23",
        title: "Colossal"
      }

      const tree = renderer.create(
          <Movie
            movie={fromJS(movie)}
          />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("<Reminder/>", () => {
    it('renders', () => {
      const reminder= {
        id: 456,
        text: "Reminder 45"
      }

      const tree = renderer.create(
          <Reminder
            reminder={fromJS(reminder)}
          />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("<CreateReminder/>", () => {
    it('renders', () => {

      const tree = renderer.create(
          <CreateReminder
            onReminderClick={ () => null }
            eventId={12}
            monthId="1"
          />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  // describe("<Nav/>", () => {
  //   it('renders', () => {
  //
  //     const tree = renderer.create(<Nav/>).toJSON();
  //
  //     expect(tree).toMatchSnapshot();
  //   });
  // });
});
