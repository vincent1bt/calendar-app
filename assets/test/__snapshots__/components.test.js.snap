// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <CreateReminder/> renders 1`] = `
<div
  className="reminder-input"
>
  <input />
  <button
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;

exports[`components <Day/> renders 1`] = `
<div>
  <section
    className="titles-day"
  >
    <h2>
      Day 
      12
      , 
      enero
    </h2>
  </section>
  <section
    className="movies"
  >
    <h2
      className="movies-title"
    >
      Movies
    </h2>
    <p
      className="no-events"
    >
      No Movies
    </p>
  </section>
  <section
    className="reminders"
  >
    <h2
      className="reminders-title"
    >
      Reminders
    </h2>
    <p
      className="no-events"
    >
      No Reminders
    </p>
    <div
      className="reminder-input"
    >
      <input />
      <button
        onClick={[Function]}
      >
        Add
      </button>
    </div>
  </section>
</div>
`;

exports[`components <Month/> renders 1`] = `
<div
  className="month"
>
  <h2>
    enero
     
    2017
  </h2>
  <ul
    className="month-names"
  >
    <li>
      <p>
        Mon
      </p>
    </li>
    <li>
      <p>
        Tue
      </p>
    </li>
    <li>
      <p>
        Wed
      </p>
    </li>
    <li>
      <p>
        Thu
      </p>
    </li>
    <li>
      <p>
        Fri
      </p>
    </li>
    <li>
      <p>
        Sat
      </p>
    </li>
    <li>
      <p>
        Sun
      </p>
    </li>
  </ul>
  <ul
    className="month-days"
  />
</div>
`;

exports[`components <Movie/> renders 1`] = `
<div
  className="movie"
>
  <div
    className="title"
  >
    <h3
      className="title-p"
    >
      Colossal
    </h3>
    <p
      className="title-over"
    >
      A woman discovers that severe catastrophic events are somehow connected to the mental breakdown from which she's suffering.
    </p>
  </div>
  <div
    className="image"
  >
    <img
      alt="Colossal"
      src="http://image.tmdb.org/t/p/w500/4VOyofBd1pexblxtDZYtYIk7NI4.jpg"
    />
  </div>
</div>
`;

exports[`components <Reminder/> renders 1`] = `
<div
  className="reminder"
>
  <p>
    Reminder 45
  </p>
</div>
`;
